src/core/eval.ml

```diff
@@ -285,19 +289,34 @@ and tree_walk : dtree -> ctxt -> stack -> (term * stack) option =
             if store then (vars.(cursor) <- add_args t args; cursor + 1)
             else cursor
           in
+          (* [default ()] carries on the matching on the default branch of the
+             tree. Nothing is added to the stack. *)
           let default () =
             let fn d =
               let stk = List.reconstruct left [] right in
               walk d stk cursor vars_id id_vars
             in
             Option.map_default fn None default
           in
+          (* [walk_binder a  b  id tr]  matches  on  binder  [b]  of type  [a]
```

Comment:
I don't like very much the spacing in `[walk_binder a  b  id tr]`. In fact, I don't like very much justified text. I prefer flush left text, as in hand writing.

Reply:
Ok, I'm fine with both.